main.py

```diff
@@ -1,9 +1,184 @@
-print("Wyświetlenie listy zrobionych zadań: 1")
-print("Wyświetlenie listy zadań do wykonania: 2")
-print("Dodaj zadanie: 3")
-while True:
-    odpowiedz = input('Wybierz menu ')
-    if odpowiedz == "3":
-        nowe_zadanie = input("Napisz jakie zadanie chcesz dodać: ")
-        print(nowe_zadanie)
-        break
\ No newline at end of file
+import sqlite3
+from task import Task
+
+task_list = []
+database = "pythonsqlite.db"
+
+
+def create_connection(db_file):
+    """ create a database connection to the SQLite database
+        specified by db_file
+    :param db_file: database file
+    :return: Connection object or None
+    """
+    conn = None
+    try:
+        conn = sqlite3.connect(db_file)
+        return conn
+    except sqlite3.Error as e:
+        print(e)
+
+    return conn
+
+
+def create_table(conn, create_table_sql):
+    """ create a table from the create_table_sql statement
+    :param conn: Connection object
+    :param create_table_sql: a CREATE TABLE statement
+    :return:
+    """
+    try:
+        c = conn.cursor()
+        c.execute(create_table_sql)
+    except sqlite3.Error as e:
+        print(e)
+
+
+def create_task(conn, task_name):
+    """
+    Create a new task
+    :param conn:
+    :param task_name:
+    :return:
+    """
+
+    sql = """ INSERT INTO tasks(name,status)
+              VALUES(?,?) """
+    cur = conn.cursor()
+    cur.execute(sql, task_name)
+    conn.commit()
+    return cur.lastrowid
+
+
+def main_db_upload(upload):
+    # create a database connection
+    conn = create_connection(database)
+
+    sql_create_tasks_table = """CREATE TABLE IF NOT EXISTS tasks (
+                                        id integer PRIMARY KEY,
+                                        name text NOT NULL,
+                                        status bool NOT NULL
+                                    );"""
+    # create tables
+    if conn is not None:
+
+        # create tasks table
+        create_table(conn, sql_create_tasks_table)
+    else:
+        print("Error! cannot create the database connection.")
+    with conn:
+        # create tasks
+        create_new_task = (upload.name, upload.status)
+        create_task(conn, create_new_task)
+
+
+def main_db_download(status):
+    # create a database connection
+    conn = create_connection(database)
+
+    sql_create_tasks_table = """CREATE TABLE IF NOT EXISTS tasks (
```

Comment:
Nie rozumiem, czemu metoda która odpowiada za wyciąganie z bazy jakiś danych, w przypadku, gdy nie ma tabeli, to tworzy takową. Łamie to zasadę żeby jakaś funkcja robiła tylko 1 rzecz.  Masz jakiś tutorial na którym się wzorowałeś? Bo może tam jakiś grubszy kontekst jest. Bo też nie kumam sensu metody main_db_download(), czemu jest tu ona i potem wywołuje kolejną metodę z selectem na bazie.

Reply:
https://www.sqlitetutorial.net/sqlite-python/sqlite-python-select/

Reply:
Faktycznie często się to powtarza, ale jak próbowałem to wyciąć i uporządkować to apka się przewróciła. Będę musiał spróbować jeszcze raz.

Reply:
Nie no w tutku jest troszkę inaczej :smile:

Reply:
Wywalić man_db_download. W miejscu gdzie go wywołujesz, czyli w `incomplete_task` i `complete_task` zmieniasz z main_db_download(True/False) na select_task_by_status(conn, True/False). A nad tym robisz conn = create_connection(database). 
Coś takiego
```python
def incomplete_task():
    print("Pobrane zadania do zrobienia z bazy danych")
    conn = create_connection(database)
    select_task_by_status(conn, False)
    return_to_menu()
```

Reply:
Bo tworzenie tabeli w wyciąganiu danych z tabeli jest słabe

Reply:
Hmmm oni wgl używają `with`
``` python
def incomplete_task():
    print("Pobrane zadania do zrobienia z bazy danych")
    conn = create_connection(database)
    with conn:
        select_task_by_status(conn, False)
    return_to_menu()
```

Reply:
Zgadza się, to pośpiech jest winowajcą, uporządkuje to sprawdzisz :D

Reply:
Zresztą create_connection też jest trochę inaczej, bo oni obudowywują to w blok `try / except`

Reply:
Można sprawdzać, uporządkowałem na szybko.

Reply:
Lepiej

Reply:
Bum, wywaliłeś metodę main_db_upload, a ona tworzyła tabele task w bazie. Teraz nic jej nie tworzy. Jak wywalisz plik z bazą to apka się wykrzaczy przy dodawaniu nowego taska

Reply:
Wiem, zastanawiam się w którym miejscu to dodać aby to wykonywało się tylko raz, a nie wtedy kiedy chcę coś z tą tabelą robić. Może w głównym main, przy starcie apki.

Reply:
Taka jest moja propozycja, jak ocenisz?

Reply:
Można by zrobić jakąs metodę prepare_db(), która by była wywoływana w main. 
I potem porobić osobne metody do tworzenia różnych tabel. 
``` python
def create_table_task(conn):
sql_create_tasks_table = """CREATE TABLE IF NOT EXISTS tasks (
                                                id integer PRIMARY KEY,
                                                name text NOT NULL,
                                                status bool NOT NULL
                                            );"""
    # create tables
    if conn is not None:
        try:
            c = conn.cursor()
            c.execute(sql_create_tasks_table)
        except sqlite3.Error as e:
            print(e)

def create_table_dupa(conn):
    ...

def prepare_db():
    conn = create_connection(database)

    create_table_task(conn)
    create_table_dupa(conn) 
    tutaj kolejne metody
```
i wtedy w __main__ wywoływać prepare_db()
Chociaż też nie do końca mi to pasi. Ale ie mam lepszego pomysłu

Reply:
Bo z TWoim podejściem, jak dojdzie nowa tabela w bazie, to trzeba będzie w __main__ dodawać kolejne wywołania metod i będzie się syf tam robił

Reply:
wprowadzono Twoje rozwiązanie